### PR TITLE
Add boilerplate for scripts

### LINKED ISSUE #75: Add license header checks to the make check target
A check that validates that every go file has a license header.

For reference check : https://github.com/prometheus/prometheus/blob/7f7b21104734fd1c7a5d2b3bc812da96cd7c35a3/Makefile.common#L110

/good-first-issue

### PR DESCRIPTION
**Pull request by justaugustus**
**What this PR does / why we need it**:
- Add boilerplate templates and scripts
- Remove unused ci/ folder
- Update copyrights

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #75

**Special notes for your reviewer**:

n.b. The verify script does not remediate issues, simply reports on them. There is work happening to add a remediation script: https://github.com/kubernetes/repo-infra/pull/82

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/apis/azureprovider/v1alpha1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // +build !ignore_autogenerated
 
 /*
+Copyright 2019 The Kubernetes Authors.
```
[the remark this answers is not in the document]
Comment (justaugustus) [reply pkg/apis/azureprovider/v1alpha1/zz_generated.deepcopy.go]: I think it could be the build tag, but I haven't had a chance to investigate yet.

**Comment by tariq1890** [comment]
@justaugustus We need to add some more logic to the `verify-boilerplate.sh` script. cluster-api AWS accounts for files like the `deepcopy.go` which is failing for us.

Check [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/7cadc4d6fc81cfd6b73b140758ae2aefe2f8038d/hack/verify_boilerplate.py#L117)

File: pkg/apis/azureprovider/v1alpha1/zz_generated.deepcopy.go [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // +build !ignore_autogenerated
```
Comment (tariq1890) [inline pkg/apis/azureprovider/v1alpha1/zz_generated.deepcopy.go]: This is the culrprit. We need to account for special cases like this. AWS is doing the same, so we should follow suit IMO

File: pkg/apis/azureprovider/v1alpha1/zz_generated.deepcopy.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tariq1890) [reply pkg/apis/azureprovider/v1alpha1/zz_generated.deepcopy.go]: Yup it is. Check my other comments.

File: pkg/apis/azureprovider/v1alpha1/zz_generated.deepcopy.go (hunk 2 above)
Comment (justaugustus) [reply pkg/apis/azureprovider/v1alpha1/zz_generated.deepcopy.go]: It's only part of the problem though.
- Their script differs from the one in k/k
- We're using different tools for testing
- I think they're using different tags/tools for generating code

Until I get a chance to investigate a little of that, I'm going to hold off on editing the script.

**Comment by justaugustus** [comment]
@tariq1890 -- This is ready to go now. 
Needed to use `/hack/boilerplate/boilerplate.generatego.txt` as the boilerplate file in `pkg/apis/apis.go`, instead of `/hack/boilerplate/boilerplate.go.txt`.

**Comment by tariq1890** [comment]
Thanks @justaugustus !

**Pull request merged**